src/storages/storage_manager.js

```diff
@@ -25,42 +29,39 @@ const MAX_OPENED_STORAGES = 1000;
  * @property {LruCache} cache
  * @ignore
  */
-class StorageManager {
+export default class StorageManager {
     /**
      * @param {T} StorageConstructor
+     * @param {Configuration} [config]
      */
-    constructor(StorageConstructor) {
+    constructor(StorageConstructor, config = Configuration.getGlobalConfig()) {
         this.StorageConstructor = StorageConstructor;
         this.name = StorageConstructor.name;
-        this.cache = cacheContainer.openCache(this.name, MAX_OPENED_STORAGES);
+        this.cache = cacheContainer.openCache(this.name, config.get('maxOpenedStorages'));
+        this.config = config;
     }
 
     /**
-     * @param {string} idOrName
+     * @param {string} [idOrName]
      * @param {object} [options]
      * @param {boolean} [options.forceCloud]
      * @return {Promise<T>}
      */
     async openStorage(idOrName, options = {}) {
-        if (!process.env[ENV_VARS.LOCAL_STORAGE_DIR] && !process.env[ENV_VARS.TOKEN]) {
-            throw new Error(`Cannot use ${this.name} as neither ${ENV_VARS.LOCAL_STORAGE_DIR} nor ${ENV_VARS.TOKEN}`
-                + ' environment variable is set. You need to set one these variables in order to enable data storage.');
-        }
-
-        const isLocal = !!(process.env[ENV_VARS.LOCAL_STORAGE_DIR] && !options.forceCloud);
+        const isLocal = !!(this.config.get('localStorageDir') && !options.forceCloud);
 
         if (!idOrName) {
             const defaultIdEnvVarName = DEFAULT_ID_ENV_VAR_NAMES[this.name];
-            idOrName = process.env[defaultIdEnvVarName];
-            if (!idOrName && isLocal) idOrName = LOCAL_ENV_VARS[defaultIdEnvVarName];
+            const defaultIdConfigKey = DEFAULT_ID_CONFIG_KEYS[this.name];
+            idOrName = this.config.get(defaultIdConfigKey);
             if (!idOrName) throw new Error(`The '${defaultIdEnvVarName}' environment variable is not defined.`);
         }
 
         const cacheKey = this._createCacheKey(idOrName, isLocal);
-
         let storage = this.cache.get(cacheKey);
+
         if (!storage) {
-            const client = isLocal ? utils.apifyStorageLocal : utils.apifyClient;
+            const client = isLocal ? this.config.createLocalStorage() : this.config.createClient();
```

Comment:
See comment in Configuration about multiple instances of client and storageLocal